aws-lc-rs/tests/rsa_test.rs

```diff
@@ -836,10 +873,146 @@ fn max_encrypt_key() {
         .expect("decrypted");
 
     assert_eq!(message, plaintext);
+
+    let pkcs1_parsed_private =
+        Pkcs1PrivateDecryptingKey::new(parsed_private_key.clone()).expect("supported key");
+    let pkcs1_parsed_public =
+        Pkcs1PublicEncryptingKey::new(parsed_public_key.clone()).expect("supported key");
+
+    let message = vec![42u8; pkcs1_parsed_public.max_plaintext_size()];
+
+    let mut ciphertext = vec![0u8; pkcs1_parsed_public.ciphertext_size()];
+    let ciphertext = pkcs1_parsed_public
+        .encrypt(&message, &mut ciphertext)
+        .expect("encrypted");
+
```

Comment:
Nit: Would it be worth doing a `assert_ne!(message, ciphertext);` here, similar to what you're doing in `rsa_test.rs`?